### DISCUSSION TITLE
os.proc pid to be killed later

### DISCUSSION THREAD

**Discussion by chpache** [lihaoyi/os-lib#132] [Q&A]
Hello there!
I quite new to mill and really like it. I have the case where I would let my user start some process with one command stop this process with another command. I would need to get the id of this proc and store it in a file to be able to stop it later. Would it be possible?

**Comment by lefou** [comment]
Yeah, that would be very useful. Unfortunately, Java 8 does not come with easy access to that information, but we currently try to keep compatible to Java 8.

But it looks like there is some support since Java 9 via [`Process.pid()`](https://docs.oracle.com/javase/9/docs/api/java/lang/Process.html#pid--).

Question is, how to handle it in a compatible way? We could use reflection to dynamically support it at runtime, but we need to reflect that in the API. Also, question is, whether this can work at all in Scala native (or Scala.js)?

Any feedback is appreciated.

**Reply by fleipold** [reply]
I think it's generally worthwhile to have a facade around process management and process ids. 

os-lib would abstract over the implementation for the different target platforms. It should be possible to implement these things on the jvm, in a native posix like environment and on node.js. The browser runtime is a different beast, but that is also true for file operations.
 
Requiring at least Java 9 shouldn't be too harsh a requirement.

**Reply by lefou** [reply]
Current os-lib is already on Java 11, so no need to consider older JVM APIs.
